From 9232b710bbd342564850c3e241b42e304349a93c Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@freescale.com>
Date: Fri, 16 Apr 2010 15:00:25 +0800
Subject: [PATCH] ENGR00122612-2 iMX23 Lower the suspend current to 9mA.

Fix LCDIF suspend issue and improve suspend flow,now
suspend current can low to 9mA with LCD panel.

Signed-off-by: Frank Li <Frank.Li@freescale.com>
---
 drivers/video/mxs/lcd_lms430.c |    6 ++++++
 drivers/video/mxs/lcdif.c      |    3 ---
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/video/mxs/lcd_lms430.c b/drivers/video/mxs/lcd_lms430.c
index 00273b7..c167774 100644
--- a/drivers/video/mxs/lcd_lms430.c
+++ b/drivers/video/mxs/lcd_lms430.c
@@ -113,11 +113,17 @@ out:
 static void release_panel(struct device *dev,
 			  struct mxs_platform_fb_entry *pentry)
 {
+	/* Reset LCD panel signel. */
+	__raw_writel(BM_LCDIF_CTRL1_RESET,
+		REGS_LCDIF_BASE + HW_LCDIF_CTRL1_CLR);
+	mdelay(100);
 	mxs_lcdif_notify_clients(MXS_LCDIF_PANEL_RELEASE, pentry);
 	release_dotclk_panel();
 	mxs_lcdif_dma_release();
 	clk_disable(lcd_clk);
 	clk_put(lcd_clk);
+	__raw_writel(BM_LCDIF_CTRL_CLKGATE,
+		     REGS_LCDIF_BASE + HW_LCDIF_CTRL_SET);
 }
 
 static int blank_panel(int blank)
diff --git a/drivers/video/mxs/lcdif.c b/drivers/video/mxs/lcdif.c
index 03d4abd..f7d48a6 100644
--- a/drivers/video/mxs/lcdif.c
+++ b/drivers/video/mxs/lcdif.c
@@ -101,9 +101,6 @@ void mxs_lcdif_stop(void)
 	__raw_writel(BM_LCDIF_CTRL_LCDIF_MASTER,
 		     REGS_LCDIF_BASE + HW_LCDIF_CTRL_CLR);
 	udelay(100);
-
-	__raw_writel(BM_LCDIF_CTRL_CLKGATE,
-		     REGS_LCDIF_BASE + HW_LCDIF_CTRL_SET);
 }
 EXPORT_SYMBOL(mxs_lcdif_stop);
 
-- 
1.5.4.4

